Lib/Rabbit4000/tcpip/DMAETH100.LIB

```diff
@@ -808,34 +817,33 @@ dma_rcv_isr::
    jr    nz,.reset_rx_port ; 5
 
    ; Check if only status rcv'd with no packet data
+   ; This is the catch all for errors with R6000
 	ld		hl,(ix+[_decb]+rcvlen)
    test	hl
    jr		z,.test_rx_error
 
    ; Check for valid Rcv Packet
-	ld		a,(ix+[_decb]+rcvlsbstat)
+   ; Don't know where this is documented for R6000?
+#if !defined(_RAB6K)
+   ld		a,(ix+[_decb]+rcvlsbstat)
    bit   7,a               ; 4
    jr    z,.test_rx_error
+#endif
 
 .test_fire_dma:
-   ; Process received packet.
-   ld    a,(ix+[_decb]+receiving)
-   ld    (ix+[_decb]+receiving),0   ; one less available
-
-   ; test whether 'receiving' was decremented to zero...
-   or    a
-   jr    z,.cont20        ; 5 - jump if it was already 0
-   ; yes, receiving was 1, therefore we need to immediately allocate a new rx
-   ; buffer and fire it off.
+   ; Good packet in
+   ld    (ix+[_decb]+receiving),0   ; technically, we're now done with this buffer and will need another
+   ; quickly try to get another buffer
+   call  dmaeth100_alloc_rx
+   jr    c,.cont21				; CF means couldn't get a buffer
 #ifdef DMAETH100_SUPERDEBUG
    ld    l,ISRL_ALLOC_2
    call  write_ilog
 #endif
-   call  dmaeth100_alloc_rx

```

Comment:
I was concerned about the interrupt log stuff crunching registers prior to the alloc.  I doubt if this thing will work with SUPERDEBUG on anyway, but that's fine.
Regarding the receiving stuff, that seemed to be a mixture of "old" code that had 2 buffers for receive operation that wasn't implemented all the way in this codebase.  You can see it in the other non-100 file.  To me, it looks like someone started with that non-100 file, tried to make it work, then pulled out the 2 buffer operation but didn't pull everything out.  In the end, it would probably be quite nice to have a pool of receive buffers  ;-)